### PR TITLE
Added options for styling when selecting/deselecting rows

### PR DESCRIPTION
**Pull request by geoffreylitt**

### CODE REVIEW

File: dist/wildcard.js [hunk 1]
```diff
@@ -129509,19 +129512,30 @@
...
+	            else {
+	                row.els.forEach(function (el) {
+	                    if (el.style) {
+	                        el.style["border"] = "solid 2px " + highlightColor;
+	                    }
+	                });
+	                row.els[0].scrollIntoView({ behavior: "smooth", block: "center" });
+	            }
+	            // Clear styling on other rows
+	            var unselectedRows = lodash.flatten(rows.filter(function (r) { return r !== row; }).map(function (r) { return r.els; }));
```
Comment (geoffreylitt) [inline dist/wildcard.js]: why

File: src/core.ts [hunk 2]
```diff
@@ -202,6 +202,16 @@ interface SiteAdapterOptions {
...
+
+  /**
+   * Styling function for the selected row
+   */
+  styleSelectedRow?(row: DataRow): void;
+
+  /**
+   * Styling function for the non-selected rows
+   */
+  styleUnselectedRows?(rows: DataRow[]): void;
```
Comment (geoffreylitt) [inline src/core.ts:214]: @TylerMillis thoughts on making this `styleUnselectedRow` and having it act on a single row, so it's more consistent with `styleSelectedRow`?

File: src/core.ts [hunk 3]
```diff
@@ -202,6 +202,16 @@ interface SiteAdapterOptions {
       (Enables special handling of certain HTML elements, which doesn't
        work well on all sites) */
   iframe?: boolean;
+
+  /**
+   * Styling function for the selected row
```
Comment (geoffreylitt) [inline src/core.ts:207]: Could you make this a tad more specific so it comes out better in the docs? eg "Applies custom styles to the selected row in the original page."

File: src/core.ts [hunk 4]
```diff
@@ -202,6 +202,16 @@ interface SiteAdapterOptions {
...
+   */
+  styleSelectedRow?(row: DataRow): void;
```
Comment (geoffreylitt) [inline src/core.ts:209]: Thoughts on passing in the list of elements here, rather than the DataRow?

Also maybe `onRowSelected` would be a more generic way to describe this function, as a "hook" that lets you add arbitrary behavior when a row is selected -- thoughts?

File: src/core.ts [hunk 5]
```diff
@@ -672,20 +682,33 @@ const createTable = (options: SiteAdapterOptions) => {
...
-        }
-      })
-      row.els[0].scrollIntoView({ behavior: "smooth", block: "center" })
+      // For multiple rows, we apply styling on the whole row
+      if(options.styleSelectedRow){
+        options.styleSelectedRow(row);
+      }
+      else {
+        row.els.forEach(el => {
+          if (el.style) {
```
Comment (geoffreylitt) [inline src/core.ts:691]: Why is this needed?

File: src/core.ts (hunk 2 above)
Comment (TylerMillis) [reply src/core.ts]: I made this function more in line with the selected row function. It makes sense for them to be more uniform and easier to use.

File: src/core.ts (hunk 4 above)
Comment (TylerMillis) [reply src/core.ts]: I switched it to passing in the list of elements since I realized I was doing that anyways with the unselected rows function. I also updated the selected row function to be onRowSelected as described. Who knows what we might want to do in the future, so making it a function that is more broad and flexible makes sense to me.

**Comment by geoffreylitt** [comment]
@TylerMillis I integrated your code into the new codebase here:
https://github.com/geoffreylitt/wildcard/commit/78d4574cfee0107726cbebda726743f558b99369

Closing this PR
